### DISCUSSION TITLE
how to clear the field when a form box is hidden

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2341] [Help]
Is there a way to clear the user input data in a field if the input box is then hidden due to a conditional formating?

I have a progressive dropdown menu I made with conditional formatting rules. the problem I am running into is, If a user inputs data in a field then selects to go with a different set of inputs, the user changes the selection in the drop-down menu thus changing the input fields shown. The original data entered in those now hidden fields are still saving to the SQL DB is there something I can do to prevent this? 
maybe in conditional formating when hidden clear. I tried when hidden disable but that did not work.

**Comment by joebudi** [comment]
Hey @AndrewJRamay 

> then selects to go with a different set of inputs

Could you explain this process in a little more detail?

**Reply by KernOil** [reply]
@joebudi 
for example the flow starts with one box 
![image](https://user-images.githubusercontent.com/80058238/129388200-836ad221-5e39-4091-998d-5e786436f46e.png)

as you select an item the corresponding next field appears 
![image](https://user-images.githubusercontent.com/80058238/129388291-3c960c24-3803-4b5c-88bf-333dcc247fef.png)

![image](https://user-images.githubusercontent.com/80058238/129388331-2c1eba58-0ec2-4283-929f-daf59fc50de8.png)

and so on 

![image](https://user-images.githubusercontent.com/80058238/129388379-7d663c65-e616-4e1b-a0ca-9c25a415d3d6.png)

the problem I am having is that if any data is inputted into any box including the drop-down menus if I am to change a selection of the drop-down menu that will, in turn, bring up different fields the data input into the now hidden fields are still saving. 

![image](https://user-images.githubusercontent.com/80058238/129388652-f4b36e78-aa15-42f0-81a0-4b9d2a144d0c.png)

now I have changed the selection but all of the previous selection is still sitting in the hidden boxes waiting for the user to hit save. then it is cleared. 

is there a way to clear the data for boxes that are not actively displayed.
